Problem: hax crashes if there is no Consul leader

`hax` crashes if there is no Consul leader:

```
Dec 29 17:24:23.462020 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:23 [INFO] consul: cluster leadership lost
Dec 29 17:24:24.073034 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:24 [WARN] agent: Check "service:47" is now warning
Dec 29 17:24:25.086275 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:25 [WARN] agent: Check "service:47" is now warning
Dec 29 17:24:26.095287 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:26 [WARN] agent: Check "service:47" is now warning
Dec 29 17:24:26.477453 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:26 [ERR] memberlist: Failed fallback ping: read tcp 172.28.128.53:47812->172.28.128.152:8301: i/o timeout
Dec 29 17:24:26.477751 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:26 [INFO] memberlist: Suspect hare64479-build-ssu1 has failed, no acks received
Dec 29 17:24:27.105178 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:27 [WARN] agent: Check "service:47" is now warning
Dec 29 17:24:28.114659 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:28 [WARN] agent: Check "service:47" is now warning
Dec 29 17:24:28.718939 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:28 [WARN]  raft: Heartbeat timeout from "" reached, starting election
Dec 29 17:24:28.719254 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:28 [INFO]  raft: Node at 172.28.128.53:8300 [Candidate] entering Candidate state in term 4
Dec 29 17:24:29.123988 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:29 [WARN] agent: Check "service:47" is now warning
Dec 29 17:24:29.477367 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:29 [ERR] memberlist: Failed fallback ping: read tcp 172.28.128.53:47814->172.28.128.152:8301: i/o timeout
Dec 29 17:24:29.477655 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:29 [INFO] memberlist: Suspect hare64479-build-ssu1 has failed, no acks received
Dec 29 17:24:29.985466 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:29 [ERR] http: Request GET /v1/kv/m0conf/nodes?recurse=1, error: No cluster leader from=127.0.0.1:44372
Dec 29 17:24:30.133318 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:30 [WARN] agent: Check "service:47" is now warning
Dec 29 17:24:30.477644 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:30 [INFO] memberlist: Marking hare64479-build-ssu1 as failed, suspect timeout reached (0 peer confirmations)
Dec 29 17:24:30.477973 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:30 [INFO] serf: EventMemberFailed: hare64479-build-ssu1 172.28.128.152
Dec 29 17:24:30.478294 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:30 [INFO] consul: Removing LAN server hare64479-build-ssu1 (Addr: tcp/172.28.128.152:8300) (DC: dc1)
Dec 29 17:24:32.343400 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:32 [ERR] agent: Coordinate update error: No cluster leader
Dec 29 17:24:32.343922 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:32 [INFO]  raft: Duplicate RequestVote for same term: 4
Dec 29 17:24:32.344274 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:32 [INFO] serf: EventMemberJoin: hare64479-build-ssu1 172.28.128.152
Dec 29 17:24:32.344530 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:32 [INFO] consul: Adding LAN server hare64479-build-ssu1 (Addr: tcp/172.28.128.152:8300) (DC: dc1)
Dec 29 17:24:32.344765 hare64479-build-ssu2 hax[3466]: 2019-12-29 17:24:32,344 [DEBUG] {qconsumer} http://127.0.0.1:8500 "GET /v1/kv/m0conf/nodes?recurse=1 HTTP/1.1" 500 41
Dec 29 17:24:32.355811 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:32 [WARN] agent: Check "service:47" is now warning
Dec 29 17:24:32.525137 hare64479-build-ssu2 hax[3466]: 2019-12-29 17:24:32,344 [ERROR] {qconsumer} **ERROR**
Dec 29 17:24:32.775139 hare64479-build-ssu2 hax[3466]: Traceback (most recent call last):
Dec 29 17:24:32.775416 hare64479-build-ssu2 hax[3466]: File "/opt/seagate/hare/lib64/python3.6/site-packages/hax/halink.py", line 16, in wrapper
Dec 29 17:24:32.775649 hare64479-build-ssu2 hax[3466]: return fn(*args, **kwargs)
Dec 29 17:24:32.775897 hare64479-build-ssu2 hax[3466]: File "/opt/seagate/hare/lib64/python3.6/site-packages/hax/halink.py", line 158, in ha_nvec_get_reply
Dec 29 17:24:32.776123 hare64479-build-ssu2 hax[3466]: n.note.no_id.f_key) != 'passing':
Dec 29 17:24:32.860240 hare64479-build-ssu2 hax[3466]: File "/opt/seagate/hare/lib64/python3.6/site-packages/hax/util.py", line 212, in get_conf_obj_status
Dec 29 17:24:32.860581 hare64479-build-ssu2 hax[3466]: node_items = self._kv_get('m0conf/nodes', recurse=True)
Dec 29 17:24:32.860900 hare64479-build-ssu2 hax[3466]: File "/opt/seagate/hare/lib64/python3.6/site-packages/hax/util.py", line 84, in _kv_get
Dec 29 17:24:32.861242 hare64479-build-ssu2 hax[3466]: return self.cns.kv.get(key, **kwargs)[1]
Dec 29 17:24:32.861559 hare64479-build-ssu2 hax[3466]: File "/opt/seagate/hare/lib/python3.6/site-packages/consul/base.py", line 554, in get
Dec 29 17:24:32.861857 hare64479-build-ssu2 hax[3466]: params=params)
Dec 29 17:24:32.862160 hare64479-build-ssu2 hax[3466]: File "/opt/seagate/hare/lib/python3.6/site-packages/consul/std.py", line 22, in get
Dec 29 17:24:32.862397 hare64479-build-ssu2 hax[3466]: self.session.get(uri, verify=self.verify, cert=self.cert)))
Dec 29 17:24:32.862614 hare64479-build-ssu2 hax[3466]: File "/opt/seagate/hare/lib/python3.6/site-packages/consul/base.py", line 223, in cb
Dec 29 17:24:32.862839 hare64479-build-ssu2 hax[3466]: CB._status(response, allow_404=allow_404)
Dec 29 17:24:32.863058 hare64479-build-ssu2 hax[3466]: File "/opt/seagate/hare/lib/python3.6/site-packages/consul/base.py", line 188, in _status
Dec 29 17:24:32.863312 hare64479-build-ssu2 hax[3466]: raise ConsulException("%d %s" % (response.code, response.body))
Dec 29 17:24:32.863535 hare64479-build-ssu2 hax[3466]: consul.base.ConsulException: 500 No cluster leader
```
http://gitlab.mero.colo.seagate.com/mero/hare/-/jobs/81864/artifacts/file/test-boot2_ssu2_syslog.txt

The Consul's leader may be lost for a few seconds sometimes due to network issues. For example, in the above case it was restored quickly:

```
Dec 29 17:24:35.270303 hare64479-build-ssu2 hare-consul[2588]: 2019/12/29 17:24:35 [INFO] consul: cluster leadership acquired
```

So `hax` should be tolerable for such cases.

--- a/hax/hax/util.py
+++ b/hax/hax/util.py
@@ -81,7 +81,11 @@ class ConsulUtil:
         we want to invoke Consul.kv.get()
         """
         assert key
-        return self.cns.kv.get(key, **kwargs)[1]
+        try:
+            return self.cns.kv.get(key, **kwargs)[1]
+        except ConsulException as e:
+            raise HAConsistencyException('Could not access Consul KV')\
+                from e
 
     def _catalog_service_get(self, svc_name: str) -> List[Dict[str, Any]]:
         try:

